Essentials/src/com/earth2me/essentials/utils/LocationUtil.java

```diff
@@ -274,76 +273,79 @@ public static Location getRoundedDestination(final Location loc) {
     public static Location getSafeDestination(final IUser user, final Location loc) throws Exception {
         return getSafeDestination(null, user, loc);
     }
-
+    
     public static Location getSafeDestination(final IEssentials ess, final IUser user, final Location loc) throws Exception {
+    	return getSafeDestination(ess, user, loc, false);
+    }
+
+    public static Location getSafeDestination(final IEssentials ess, final IUser user, final Location loc, final boolean overrideCenter) throws Exception {
         if (user.getBase().isOnline() && loc.getWorld().equals(user.getBase().getWorld()) && (user.getBase().getGameMode() == GameMode.CREATIVE || user.isGodModeEnabled()) && user.getBase().getAllowFlight()) {
             if (shouldFly(loc)) {
                 user.getBase().setFlying(true);
             }
             // ess can be null if old deprecated method is calling it.
             System.out.println((ess == null) + " " + ess.getSettings().isTeleportToCenterLocation());
-            if (ess == null || ess.getSettings().isTeleportToCenterLocation()) {
+            if (ess == null || (ess.getSettings().isTeleportToCenterLocation() && !overrideCenter)) {
                 return getRoundedDestination(loc);
             } else {
                 return loc;
             }
         }
-        return getSafeDestination(loc);
+        return overrideCenter ? getSafeDestination(loc, false) : getSafeDestination(loc);
     }
-
+    
     public static Location getSafeDestination(final Location loc) throws Exception {
+    	return getSafeDestination(loc, true);
+    }
+
+    public static Location getSafeDestination(final Location loc, boolean center) throws Exception {
```

Comment:
This is a bit pedantic, but this method will fail if given a location on the corner with center being false. The previous behavior of this method used to assume that the player would only ever be in the center of the block. Now, when center is false, the player may occupy multiple blocks at a time. Checks whether or not the blocks are air aren't that big of a deal (immediately thinking about it) but checking for dangerous blocks becomes way more difficult - just because the block the player is majorly located in doesn't contain lava doesn't mean that an adjacent block contains lava which overlaps with the player's bounding box (thus burning them).